src/main/scala/com/timeout/KinesisGraphStage.scala

```diff
@@ -81,18 +105,24 @@ class KinesisGraphStage[A : ToPutRecordsRequest](putRecords: PutRecords, streamN
       recordsInFlight = dataToPush.size
 
       Future {
-        // everything in here happens in the worker thread
-        val request = new PutRecordsRequest()
-          .withRecords(dataToPush.map(_.toRequestEntry).asJava)
-          .withStreamName(streamName)
+        val maxSize = 0
+
+        val grouppedData: List[List[PutRecordsRequestEntry]] = splitBySize(dataToPush.map(_.toRequestEntry))
 
         def incrementalBackoff(err: Throwable, n: Int): Unit = {
-          val waitSec = Math.pow(2, n).toInt
-          log.error(s"Error while trying to push to Kinesis: $err.\nBacking off for $waitSec seconds")
+          val waitSec = Math.pow(2, Math.max(n, 12)).toInt // Don't not wait longer than 4096 seconds between two next tries
```

Comment:
One hour delay between next tries is IMO long enough. Probably we should throw an error if it doesn't succeed within 10 tries anyway. Plus if the request is buggy, e.g. is too big, then retrying won't help so we could check the error message and fail (throw an exception? or return error for all).

Reply:
the retry max was set to 7.

Reply:
Oh ok my bad, I didn't notice that. Thanks for pointing it out.